ec/src/lib.rs

```diff
@@ -214,6 +214,8 @@ pub trait ProjectiveCurve:
         self = res;
         self
     }
+
+    fn get_x(&mut self) -> &mut Self::BaseField;
```

Comment:
I don't think this is well defined for projective curve representations, as the same curve point can have different projective coordinates, which means that the X coordinate can differ. 

(Also no code in this PR uses it, but I guess your later PRs will?)

Reply:
Hmm yes, I can remove it for now, it will be used only for a generic script to generate parameters. 

The purpose is to mutate x, in the use case for the purpose of applying an endomorphism, for which the particular Z coordinate does not matter.

Reply:
I see, maybe it would be better to add `xyz_endo_in_place`/`xyz_endo` methods in the appropriate location (in the ModelParameters trait, perhaps?)

Reply:
possibly, but the method already defined later on would apply multiplication by a constant endo param (`P::glv_endomorphism_in_place(&mut p.x);`). 

This more generic method allows you to do whatever you want with x. I don't think this is an issue in itself if you know what you're doing... If you're worried about clutter, well, unfortunately I can't think of a workaround.

Further, as far as the curves we have are concerned, `get_x` and `get_y` for the ProjectiveCurve trait seem fairly reasonable...

Reply:
But we can discuss it further in the PR for which it is required.

Reply:
Ah I meant that instead one would do something like `P::glv_endomorphism_in_place(&mut p);`, which fully encapsulates the details of the endomorphism; the signature would be something like

```rust
trait SWModelParameters {
	/// Existing stuff
	
	fn glv_endomorphism_in_place(p: &mut GroupProjective<Self>) {
		// do whatever
	}
}
```

Reply:
But yeah, we can discuss it in the GLV PR

Reply:
Just checking, the plan is to remove it from this PR right?